### PR TITLE
Make preflight request header check case-insensitive

### PR DESCRIPTION
**Pull request by Norgerman**
In jquery ajax(maybe in old versions), the `Access-Control-Request-Headers` in `OPTIONS` request value is lower case and will contains `accept` header which is already in `SimpleRequestHeaders`

### CODE REVIEW

**Comment by Eilon** [comment]
Logged a bug to track this: https://github.com/aspnet/CORS/issues/32

@kichalla can you review this PR?

File: src/Microsoft.AspNet.Cors.Core/CorsService.cs [hunk 1]
```diff
@@ -97,7 +97,8 @@ public virtual void EvaluatePreflightRequest(HttpContext context, CorsPolicy pol
 
             if (!policy.AllowAnyHeader &&
                 requestHeaders != null &&
-                !requestHeaders.All(header => policy.Headers.Contains(header, StringComparer.Ordinal)))
+                !requestHeaders.All(header => CorsConstants.SimpleRequestHeaders.Contains(header, StringComparer.OrdinalIgnoreCase) ||
+                                              policy.Headers.Contains(header, StringComparer.OrdinalIgnoreCase)))
```
Comment (kichalla) [inline src/Microsoft.AspNet.Cors.Core/CorsService.cs]: Is this change correct? This is supposed to be case-sensitive for any header other than non-simple + content-type header, right?...like can you add a test where the header in the request is "version" and the policy is "Version"...this should not match..

**Comment by kichalla** [comment]
:watch:

File: src/Microsoft.AspNet.Cors.Core/CorsService.cs (hunk 1 above)
Comment (Norgerman) [inline src/Microsoft.AspNet.Cors.Core/CorsService.cs]: I have tested this use jquery ajax in chrome, `Access-Control-Request-Headers` are always in lower case.

**Comment by Norgerman** [comment]
I have tested this use jquery ajax in chrome, Access-Control-Request-Headers are always in lower case. So the server is not able to get Access-Control-Request-Headers values like Version or verSion, the value will always be version. Thanks for your review.

**Comment by kichalla** [comment]
:shipit:

**Comment by kichalla** [comment]
commit 38728a6bcd11bfe2c5f9c1b17fde1892bf499553

Thanks for the contribution! @Norgerman

**Comment by Norgerman** [comment]
:thumbsup:

**Pull request closed without merging**